### PR TITLE
✨ Error box UI to show which engines failed with several UI improvements 

### LINKED ISSUE #185: ✨ Error box UI to show which engines failed
Add an Error box in the Search page to show which engines failed and why they failed by providing the `error-type`. Also show the severity of the error by providing different color for each error type. 

The error box UI entries would look something like this:
```
Engine Name     (Error Type)      [Color for Severity]
```

### PR DESCRIPTION
**Pull request by neon-mmd**
## What does this PR do?

This PR provides an error-box for displaying which engines failed while fetching results from. It also provides an error messages on pages where no results were found. Moreover, it also provides a heading in the navbar to allow user to navigate to the homepage easily.

## Why is this change important?

This change is essential as it allows the user to know which engines failed and why, which in turn allows the user to report errors/bugs/issues easily. Also, it provides the user with an error message when no results were found for the particular search query. It also, provides a heading in the navbar to allow user to navigate to the homepage easily which both provides a better user experience to the user.

## How to test this PR locally?

It can be tested by installing and running `Websurfx` as mentioned in the `docs` and on the `readme` and by launching the browser and thoroughly testing. By searching the current page and intentionally causing an error and checking whether the dropdown works or not and whether icon changes or not in the search page. Also, checking whether the `websurfx` heading in the navbar navigates to the homepage or not. Moreover, checking whether the page which provided no results also provides a kind error message to the user. 

```
http://127.0.0.1:8080/search?q=ls&page=100000000
```

## Author's checklist

- [x] add a dropdown error box button in the search page near the search bar.
- [x] provide an error message to the user when no results were found.
- [x] provide a heading in the navbar to allow user to navigate to the homepage.

## Related issues

Closes #185

### CODE REVIEW

File: public/templates/search_bar.html [hunk 1]
```diff
@@ -1,9 +1,2 @@
-<div class="search_bar">
-  <input
-    type="search"
-    name="search-box"
-    value="{{this.pageQuery}}"
-    placeholder="Type to search"
-  />
-  <button type="submit" onclick="searchWeb()">search</button>
+{{>bar this}}
```
Comment (xffxff) [inline public/templates/search_bar.html:1]: Is this file still useful?

**Review by xffxff — DISMISSED** [review]
Really cool! 
(but this gif is a bit blurry :joy:)

image

File: public/templates/search_bar.html (hunk 1 above)
Comment (neon-mmd) [reply public/templates/search_bar.html:1]: Yes, but now that I think about it carefully, I feel that maybe I can remove this file and maybe edit the index.html file instead. So thanks :heart: for taking the time to review this PR, I really appreciate it :+1:  and thanks for pointing out this as well, I will make the change right away :slightly_smiling_face: .

**Comment by neon-mmd** [comment]
> Really cool! (but this gif is a bit blurry 😂)
> 
> image

Thanks for pointing this out as well, I will make the change right away :slightly_smiling_face: .

**Review by xffxff — APPROVED** [review]

**Pull request merged**